distributed/core.py

```diff
@@ -316,7 +342,39 @@ def __init__(
         connection_args=None,
         timeout=None,
         io_loop=None,
+        local_directory=None,
     ):
+        from distributed.nanny import Nanny
+
+        if local_directory is None:
+            local_directory = (
+                dask.config.get("temporary-directory") or tempfile.gettempdir()
+            )
+
+        if "dask-worker-space" not in str(local_directory):
+            local_directory = os.path.join(local_directory, "dask-worker-space")
+
+        self._original_local_dir = local_directory
+
+        with warn_on_duration(
+            "1s",
+            "Creating scratch directories is taking a surprisingly long time. ({duration:.2f}s) "
+            "This is often due to running workers on a network file system. "
+            "Consider specifying a local-directory to point workers to write "
+            "scratch data to a local disk.",
+        ):
+            self._workspace = WorkSpace(local_directory)
+
+            if isinstance(self, Nanny):
```

Comment:
I see why this is here, but it's pretty unfortunate from a class hierarchy perspective. Do you think this logic could be reasonably moved into `Nanny.__init__`?

Reply:
Wasn't happy with it either; but think maybe https://github.com/dask/distributed/pull/7802/commits/04d8ea537a6211a1bc98418f42255df0fedbe2a4 is a better solution?